core/src/com/unciv/ui/tilegroups/CityButton.kt

```diff
@@ -70,13 +70,14 @@ class CityButton(val city: CityInfo, internal val tileGroup: WorldTileGroup, ski
             // clicking swings the button a little down to allow selection of units there.
             // this also allows to target selected units to move to the city tile from elsewhere.
             if (isButtonMoved) {
+                val currentCiv = UncivGame.Current.worldScreen.viewingCiv
                 // second tap on the button will go to the city screen
                 // if this city belongs to you
                 if (UncivGame.Current.viewEntireMapForDebug || belongsToViewingCiv()) {
                     UncivGame.Current.setScreen(CityScreen(city))
-                } else {
+                } else if (currentCiv.knows(city.civInfo)) {
                     // If city doesn't belong to you, go directly to its owner's diplomacy screen.
-                    val screen = DiplomacyScreen(UncivGame.Current.worldScreen.viewingCiv)
+                    val screen = DiplomacyScreen(currentCiv)
                     screen.updateRightSide(city.civInfo)
                     UncivGame.Current.setScreen(screen)
                 }
```

Comment:
I think we shouldn't let the button move either.  In fact, probably have this check as the first line of setButtonActions()

```kotlin
val currentCiv = UncivGame.Current.worldScreen.viewingCiv
if (!belongsToViewingCiv() && !currentCiv.knows(city.civInfo)) return
```

Reply:
Moving the button is useful in order to see the units which can be overlapped by the button.

Reply:
Under what circumstance can you see the city button of a civ you don't know? If you can see the tile, you should 'meet' the civilization, no?